### ISSUE TITLE
What is the default value of a requests client?

### ISSUE THREAD

**Issue by bzbarsky** [whatwg/fetch#907]
Spec at https://fetch.spec.whatwg.org/#concept-request-client says:

> A request has an associated client (null or an environment settings object). 

There is no "unless stated otherwise" clause, but there is also no indication what the ```client``` should be set to when a request is created.  I assume it should default to `null`, but it's not 100% clear.

**Comment by bzbarsky** [comment]
@annevk

**Comment by annevk** [comment]
The expectation is that everyone that creates a request sets this field to a value. In a way this might be a duplicate of #186.

**Comment by bzbarsky** [comment]
Would it make sense to require this value to create a request at all, then?  That is the "create a request" algorithm would take it as input.  That would make it a lot easier to catch cases where people are failing to set it, I'd think.

**Comment by annevk** [comment]
Yeah, I hadn't really created such abstractions yet since not everyone adopted Fetch as their infrastructure (e.g., CSS).

We might have enough usage now though to figure out how to refactor various aspects.
